sharding/indexHashedNodesCoordinatorRegistry.go

```diff
@@ -240,3 +245,23 @@ func serializableValidatorArrayToValidatorArray(sValidators []*SerializableValid
 
 	return result, nil
 }
+
+// NodesInfoToValidators maps nodeInfo to validator interface
+func NodesInfoToValidators(nodesInfo map[uint32][]*NodeInfo) (map[uint32][]Validator, error) {
+	validatorsMap := make(map[uint32][]Validator)
+
+	for shId, nodeInfoList := range nodesInfo {
+		validators := make([]Validator, 0)
+		for _, nodeInfo := range nodeInfoList {
+			validator, err := NewValidator(nodeInfo.PubKey(), nodeInfo.Address())
+			if err != nil {
+				return nil, err
```

Comment:
Are we sure we want to break the loop here (or skip instead)?

Reply:
yes